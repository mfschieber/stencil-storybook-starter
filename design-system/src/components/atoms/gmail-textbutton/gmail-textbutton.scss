.textbutton {
  width: fit-content;
  height: 100%;
  height: fit-content;
  color: black;
  background: white;
  border-color: #d0d0d0;
  border-width: 0.4px;
  border-radius: 0.5rem;

  &.true {
    color: #adadad;
    background-color: #f1f1f1;
  }

  p {
    /* Sizes */
    &.large {
      margin: 8px 12px;
      font-weight: 900;
      font-size: 32px;
    }
    &.medium {
      margin: 6px 9px;
      font-weight: 700;
      font-size: 26px;
    }
    &.small {
      margin: 4px 6px;
      font-weight: 600;
      font-size: 20px;
    }
    &.xsmall {
      margin: 2px 3px;
      font-weight: 400;
      font-size: 14px;
    }
    &.normal {
      font-weight: normal;
    }
    &.italic {
      font-weight: normal;
      font-style: italic;
    }
    &.bold {
      font-weight: bold;
    }
  }
}

.textbutton:hover {
  background-color: rgb(226, 222, 222);
  &.true {
    color: #adadad;
    background-color: #f1f1f1;
  }
}
