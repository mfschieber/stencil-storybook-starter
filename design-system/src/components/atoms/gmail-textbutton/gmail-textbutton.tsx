import { Component, ComponentInterface, Event, EventEmitter, h, Host, Prop, State } from '@stencil/core';

import { Identifier } from '../../shared/identifier';

@Component({
  tag: 'gmail-textbutton',
  styleUrl: 'gmail-textbutton.scss',
  shadow: true,
})
export class GmailTextbutton implements ComponentInterface {
  @Prop() identifier?: Identifier = 'textbutton';
  @State() srcIcon = '';
  @Prop() size?: 'large' | 'medium' | 'small' | 'xsmall' = 'medium';
  @Prop() type?: 'normal' | 'bold' | 'italic';
  @Prop() content?: string;

  @Event({
    eventName: 'textbuttonClicked',
    bubbles: true,
    composed: true,
  })
  textbuttonClicked: EventEmitter<Identifier>;

  textbuttonClickedHandler() {
    this.textbuttonClicked.emit(this.identifier);
  }

  render() {
    return (
      <Host>
        <button class="textbutton" onClick={this.textbuttonClickedHandler.bind(this)}>
          <p class={`p ${this.size} ${this.type}`}>{this.content}</p>
        </button>
      </Host>
    );
  }
}
