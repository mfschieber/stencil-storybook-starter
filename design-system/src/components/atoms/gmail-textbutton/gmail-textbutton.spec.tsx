import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';

import { GmailTextbutton } from './gmail-textbutton';

describe('ATOMS/gmail-textbutton', () => {
  let page;

  it('should render', async () => {
    page = await newSpecPage({
      components: [GmailTextbutton],
      template: () => <gmail-textbutton>Test</gmail-textbutton>,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should fire a click event', async () => {
    page = await newSpecPage({
      components: [GmailTextbutton],
      html: `<gmail-textbutton></gmail-textbutton>`,
    });

    const onClick = jest.fn();
    page.root.addEventListener('click', onClick);
    await page.waitForChanges();
    page.root.click();
    expect(onClick).toHaveBeenCalled();
  });
});
