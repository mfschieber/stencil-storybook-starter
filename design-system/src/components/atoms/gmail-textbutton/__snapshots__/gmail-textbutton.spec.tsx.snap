// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ATOMS/gmail-textbutton should render 1`] = `
<gmail-textbutton>
  Test
  <button class="textbutton">
    <p class="medium p undefined"></p>
  </button>
</gmail-textbutton>
`;
